taskcluster/scripts/toolchain/external-gradle-dependencies.sh

```diff
@@ -48,7 +48,19 @@ if [[ $WORKING_DIR == ${ANDROID_COMPONENTS_DIR}* ]]; then
   done
 fi
 
-./gradlew $GRADLE_ARGS $GRADLE_COMMANDS
+# gradle occasionally hangs, so run a watchdog that'll exit after 30min
+(
+sleep 30m &
+trap "kill $!" EXIT
```

Comment:
Is $! ("process ID of the most recent background command executed from the current shell") correct here? I tried testing this (simplified and with diagnostics and shorter time-outs...I may have messed something up) and, for a hung gradle, the kill resulted in "no such process" and left the hung process running.

Reply:
Hmm, you're right.  The trap in the subshell is right but the parent shell should then kill gradle.